### ISSUE TITLE
`h w` vs. `w h`

### ISSUE THREAD

**Issue by m-bock** [csicar/purescript-sized-matrices#5]
When using the library one thing caused a bit of confusion:

`newtype Matrix h w a`

I would have rather expected the arguments to be `w h a`. As I'd say `width * height` is more common. Also in vectors we follow the `x y ..` order. E.G. `fill` in this library uses this order as well.

I guess the reason for this was that it resembles the underlying implementation. But this could be independent.

**Comment by csicar** [comment]
The reason for `h w` is, that in mathematical notation this order is chosen (https://en.wikipedia.org/wiki/Matrix_(mathematics)#Size) .
I don't know why and it also seems inconsistent to the order chosen in most other mathematical notations, as you pointed out.

As you saw, I also got them mixed up when writing the `fill` function. 

I would suggest changing the implementation of `fill` (by flipping the arguments) rather than flipping the arguments in `Matrix`

**Comment by m-bock** [comment]
Ok, I see! Then I also prefer to stay with this notation.

**Comment by m-bock** [comment]
Maybe it makes sense to use `i` and `j` instead of `y` and `x`

**Comment by csicar** [comment]
Yes, you're right. Using `x` and `y` makes it even more confusing. I also had the idea to change the signature to `fill :: ... ({i, j} -> a) -> Matrix h w a`, but I don't think that's worth it

**Comment by m-bock** [comment]
Interesting. With the record you'd avoid the question of the order. Nothing prevents you from destructuring like `{j, i}` then. But I think this is another source of confusion as there is an actual order ("row first").

I mean, instead of `Int -> Int` we could use a `Vec D2 Int`...

**Comment by m-bock** [comment]
Maybe it's fine for `fill`, but the question arises when thinking of a `FunctorWithIndex` instance: Which datatype to use for the index?

**Comment by artemisSystem** [comment]
I think a `{width :: Int, height :: Int}` would be best. Or maybe just use `w` and `h`

**Comment by artemisSystem** [comment]
Nevermind, that doesn't make sense. It should be `{x, y}` or `{i, j}`

**Comment by csicar** [comment]
`{ column :: Int, row :: Int }` would be the least confusing, but also quite long.

**Comment by artemisSystem** [comment]
what about c and r?

**Comment by csicar** [comment]
I can see that working. @thought2 what's your opinion on that?
